benches/chained_spawn.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use criterion::*;
+
+mod yatp_callback {
+    use criterion::*;
+    use std::sync::mpsc;
+    use yatp::task::callback::Handle;
+
+    pub fn chained_spawn(b: &mut Bencher<'_>, iter_count: usize) {
+        let pool = yatp::Builder::new("chained_spawn").build_callback_pool();
+
+        fn iter(handle: &mut Handle<'_>, done_tx: mpsc::SyncSender<()>, n: usize) {
+            if n == 0 {
+                done_tx.send(()).unwrap();
+            } else {
+                handle.spawn(move |h: &mut Handle<'_>| {
+                    iter(h, done_tx, n - 1);
+                })
+            }
+        }
+
+        let (done_tx, done_rx) = mpsc::sync_channel(1000);
+
+        b.iter(move || {
+            let done_tx = done_tx.clone();
+            pool.spawn(move |h: &mut Handle<'_>| {
+                iter(h, done_tx, iter_count);
+            });
+
+            done_rx.recv().unwrap();
+        });
+    }
+}
+
+mod yatp_future {
+    use criterion::*;
+    use std::sync::mpsc;
+    use yatp::task::future::TaskCell;
+    use yatp::Remote;
+
+    pub fn chained_spawn(b: &mut Bencher<'_>, iter_count: usize) {
+        let pool = yatp::Builder::new("chained_spawn").build_future_pool();
+
+        fn iter(remote: Remote<TaskCell>, done_tx: mpsc::SyncSender<()>, n: usize) {
+            if n == 0 {
+                done_tx.send(()).unwrap();
+            } else {
+                let s2 = remote.clone();
+                remote.spawn(async move {
+                    iter(s2, done_tx, n - 1);
+                });
+            }
+        }
+
+        let (done_tx, done_rx) = mpsc::sync_channel(1000);
+
+        b.iter(move || {
+            let done_tx = done_tx.clone();
+            let remote = pool.remote().clone();
```

Comment:
Now `ThreadPool::remote()` returns `Remote`. I think we can change it to `&Remote`.